core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,112 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import jenkins.model.Messages;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ *
+ * @since TODO
+ */
+public abstract class ItemCategory implements ModelObject, ExtensionPoint {
```

Comment:
> Should it moves outside of `jenkins.model` package?

There is all kinds of junk in `hudson.model` with no apparent relationship. This at least is related to `[TopLevel]Item`, which is a `ModelObject`, so I think the package is fine.